src/Models/Environment.php

```diff
@@ -650,6 +650,26 @@ public function importFiles($url)
         );
     }
 
+    /**
+     * Builds a Workflow to set the primary domain for this environment.
+     *
+     * @param string $domain A domain name attached to this environment.
+     *
+     * @return Workflow
+     */
+    public function setPrimaryDomain($domain)
```

Comment:
This functionality should be in multiple functions so that its use is perfectly clear, preferably in a separate model belonging to Environment. You may want to take this functionality and create a new model, like `src/Models/Solr.php` does except you'll not use AddOnModel but instead implement `EnvironmentInterface` and `EnvironmentTrait`.

Reply:
> This functionality should be in multiple functions so that its use is perfectly clear
The function is currently 10 lines and 2 statements; the first statement sets local variable `$workflow` and the second returns it. It could be condensed to a single return statement. My point is just that I'm actually at a loss to think of how I can really break this up into more discrete functions/steps. Can you be more clear what you want with that?